### PR TITLE
Add License metrics

### PR DESCRIPTION
**Pull request by kcbark**
Metrics to monitor vdom license usage. Should be possible to add other license statuses as well.

### CODE REVIEW

File: probe.go [hunk 1]
```diff
@@ -630,6 +629,50 @@ func probeHAStatistics(c FortiHTTP) ([]prometheus.Metric, bool) {
...
+	var (
+		vdomInfo = prometheus.NewDesc(
+			"fortigate_license_vdom_info",
+			"Info metric regarding vdom licenses",
+			[]string{"type"}, nil,
+		)
+	)
+
+	type LicenseStatus struct {
+		Vdom struct {
```
Comment (bluecmd) [inline probe.go:642]: Nit: s/Vdom/VDOM (as per https://github.com/golang/go/wiki/CodeReviewComments#initialisms)

File: probe.go [hunk 2]
```diff
@@ -630,6 +629,50 @@ func probeHAStatistics(c FortiHTTP) ([]prometheus.Metric, bool) {
...
+			Used       int    `json:"used"`
+			Max        int    `json:"max"`
+		} `json:"vdom"`
+	}
+
+	type LicenseResponse struct {
+		HTTPMethod string        `json:"http_method"`
+		Results    LicenseStatus `json:"results"`
+		VDOM       string        `json:"vdom"`
+		Path       string        `json:"path"`
```
Comment (bluecmd) [inline probe.go:654]: You can omit the fields you are not using if you wish. Up to you :-)

File: README.md [hunk 3]
```diff
@@ -18,6 +18,7 @@ Global:
  * `fortigate_cpu_usage_ratio`
  * `fortigate_memory_usage_ratio`
  * `fortigate_current_sessions`
+ * `fortigate_license_vdom_info`
```
Comment (bluecmd) [inline README.md:21]: What do you think about having `fortigate_license_vdom_usage` and `fortigate_license_vdom_max` instead?
My thinking is that it makes it easier to discover, and currently it kind of breaks the [following](https://prometheus.io/docs/practices/naming/#metric-names) best-practice section:

> As a rule of thumb, either the sum() or the avg() over all dimensions of a given metric should be meaningful (though not necessarily useful). If it is not meaningful, split the data up into multiple metrics. For example, having the capacity of various queues in one metric is good, while mixing the capacity of a queue with the current number of elements in the queue is not.

Doing a `sum(fortigate_license_vdom_info)` doesn't make much sense as it contains both `max` and `usage`

File: testdata/license-status.jsonnet [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (bluecmd) [inline testdata/license-status.jsonnet:1]: Please add a comment in the first row of what URL one can use to collect the API request from. See the other jsonnet files for example.

**Review by bluecmd — COMMENTED** [review]
Cool! Thanks for your contribution!
I have some comments.

File: README.md (hunk 3 above)
Comment (kcbark) [reply README.md:21]: Yeah, makes more sense. 👍

File: probe.go [hunk 5]
```diff
@@ -630,6 +629,47 @@ func probeHAStatistics(c FortiHTTP) ([]prometheus.Metric, bool) {
 	return m, true
 }
 
+func probeLicenseStatus(c FortiHTTP) ([]prometheus.Metric, bool) {
+	var (
+		vdomUsed = prometheus.NewDesc(
+			"fortigate_license_vdom_usage",
+			"Info metric regarding used vdom licenses",
```
Comment (bluecmd) [inline probe.go:636]: Update the description of these metrics. Maybe something like "The amount of VDOM licenses currently used"  and "The total amount of VDOMs licenses available"

**Review by bluecmd — CHANGES_REQUESTED** [review]

**Comment by bluecmd** [comment]
Looks good, only the description left I'd say :-)

File: probe.go (hunk 5 above)
Comment (kcbark) [reply probe.go]: Yeah, that's a better wording, thanks :)

**Comment by kcbark** [comment]
All good? :)

**Review by bluecmd — APPROVED** [review]

**Comment by bluecmd** [comment]
🎉 Thanks!

**Pull request merged**